glottolog3/views.py

```diff
@@ -261,11 +261,71 @@ def quicksearch(request):
     return {'message': message, 'params': params, 'languoids': languoids,
         'map': map_, 'countries': countries}
 
+# ENDPOINT ADDED BY BLUEPRINT
+def bpsearch(request):
+    message = None
+    query = DBSession.query(Languoid)
+    term = request.params['bpsearch'].strip().lower()
+    params = {
+            'name': term,
+            'namequerytype': request.params['namequerytype'],
+            'multilingual': 'multilingual' in request.params
+            }
+
+    if not term:
+        query = None
+    elif len(term) < 3:
+        query = None
+        message = ('Please enter at least four characters for a name search '
+            'or three characters for an iso code')
+    elif len(term) == 8 and GLOTTOCODE_PATTERN.match(term):
+        query = query.filter(Languoid.id == term)
+        kind = 'Glottocode'
```

Comment:
this case only works for when there's a full glottocode being searched, when we should support partial glottocode (e.g. `stan1`)